docs/adrs/0015-webapps-qa-strategy.md

```diff
@@ -0,0 +1,72 @@
+# Implementing QA-Only Mode for Web Apps to Test Features
+
+* **Status**: Accepted
+* **Deciders**: Yashika Khurana, Jared Lockhart, Benjamin Forehand
+* **Date**: 2024-06-14
+
+## Context and Problem Statement
+We have identified a significant gap in our QA testing process. Bugs are often not detected until features are live in production, despite thorough testing and QA in staging environments. This issue arises from differences in data and user interaction patterns between staging and production environments. To address this, we require a method to test features in production without affecting real users.
+
+## Decision Drivers
+- **Accuracy of Testing**: Ensuring features are tested under realistic conditions to detect bugs that may not appear in staging.
+- **User Safety**: Preventing potentially disruptive features from impacting real users.
+- **Efficiency**: Streamlining the testing process to facilitate rapid and effective QA cycles.
+
+## Considered Options
+- **QA-Only Preview Collection**: Introducing a new preview collection exclusively for QA testing in the production environment.
+- **New QA Flag on Experimenter**: Implementing a `QA only` checkbox on the summary page for targeting purposes.
+
+## Decision Outcome
+Chosen option: "QA-Only Preview Collection," because it allows the creation of a separate preview environment (`nimbus-web-preview`) where QA can test features without impacting real users. This method offers a seamless transition from preview to launch without requiring changes to the recipe.
+
+## Positive Consequences
+- **Effective Validation**: Features are tested in the actual production environment, ensuring all configurations and data interactions are as expected.
+- **Quick Issue Resolution**: Allows for the rapid detection and rectification of issues before they impact users.
+- **Streamlined Processes**: Does not require cloning the experiment/rollout again once QA has tested it.
+- **Consistency**: No changes are required in the recipe from preview to launch, maintaining consistency across testing and deployment phases.
+
+## Negative Consequences
+- **Resource Requirements**: Requires the setup of a separate new collection and the successful management of multiple collections by Cirrus.
+
+## Pros and Cons of the Options
+
+### QA-Only Preview Collection
```

Comment:
Are there any effects or work on the analysis side? Presumably it wouldn't be a lot of resources to analyze the one or two people enrolled, but should Jetstream look for this flag explicitly to avoid having to do anything?

Reply:
that's a really good point, will it be helpful to have this information send as a key?

Reply:
Yea I think it would make sense to include it in the API, that way Jetstream could decide whether to consume it or not. Probably not too important for this particular doc, so up to you how you want to document that task.

Reply:
Sure I have taken a note and make sure to include this in the epic